ui/src/components/executions/date-select/DateFilter.vue

```diff
@@ -65,102 +65,117 @@
-                type: Boolean, 
-                default: false
-            }
-        },
-        emits: ["update:isRelative", "update:filterValue"],
-        created() {
-            this.selectedFilterType = this.absolute
-                ? this.filterType.ABSOLUTE
-                : this.$route.query.startDate || this.$route.query.endDate
-                    ? this.filterType.ABSOLUTE
-                    : this.filterType.RELATIVE;
-        },
-        mounted() {
-            this.$emit(
-                "update:isRelative",
-                this.selectedFilterType === this.filterType.RELATIVE,
-            );
-        },
-        data() {
-            return {
-                selectedFilterType: undefined,
-                filterType: {
-                    RELATIVE: "REL",
-                    ABSOLUTE: "ABS",
-                },
-            };
-        },
-        computed: {
-            timeRange() {
-                return this.$route.query.timeRange
-                    ? this.$route.query.timeRange
-                    : "P30D";
-            },
-            startDate() {
-                return this.$route.query.startDate
-                    ? this.$route.query.startDate
-                    : this.$moment(this.endDate).add(-30, "days").toISOString(true);
-            },
-            endDate() {
-                return this.$route.query.endDate
-                    ? this.$route.query.endDate
-                    : undefined;
-            },
-        },
-        methods: {
-            onSelectedFilterType() {
-                const relativeFilterSelected =
-                    this.selectedFilterType === this.filterType.RELATIVE;
-
-                this.$emit("update:isRelative", relativeFilterSelected);
-
-                this.tryOverrideAbsFilter(relativeFilterSelected);
-            },
-            onAbsFilterChange(event) {
-                const filter = {
-                    startDate: event.startDate,
-                    endDate: event.endDate,
-                    timeRange: undefined,
-                };
-                this.updateFilter(filter);
-            },
-            onRelFilterChange(event) {
-                const filter = {
-                    startDate: undefined,
-                    endDate: undefined,
-                    timeRange: event.timeRange,
-                };
-                this.updateFilter(filter);
-            },
-            updateFilter(filter) {
-                this.$emit("update:filterValue", filter);
-            },
-            tryOverrideAbsFilter(relativeFilterSelected) {
-                if (
-                    relativeFilterSelected &&
-                    (this.$route.query.startDate || this.$route.query.endDate)
-                ) {
-                    const forcedDefaultRelativeFilter = {timeRange: undefined};
-                    this.onRelFilterChange(forcedDefaultRelativeFilter);
-                }
-            },
-        },
+    type FilterValue = {
```

Comment:
small perf nitpick, type used to be lower to parse than interfaces for TypeScript. so if they are not necessary, we prefer using them to using types.

```suggestion
    interface FilterValue = {
```